lib/collection/src/collection_manager/optimizers/segment_optimizer.rs

```diff
@@ -39,6 +40,11 @@ pub struct OptimizerThresholds {
     pub indexing_threshold_kb: usize,
 }
 
+#[derive(Debug, Clone)]
+pub struct DefragmentationConfig {
+    pub keys: Vec<String>,
+}
+
```

Comment:
I don't think we want a separate type for this. It may be better to put it in the optimizers configuration instead. Just to prevent adding a million different config types.

@generall thoughts? Also, how do you feel about exposing this as part of a collection config?